docs/book/src/reference/metrics.md

```diff
@@ -3,6 +3,53 @@
 By default, controller-runtime builds a global prometheus registry and
 publishes a collection of performance metrics for each controller.
 
+## Exporting Metrics for Prometheus
+
+The project can be enabled to be configured by using `MonitorService` resources from
+the Prometheus Operator. Following the steps to do this setup.
+
+1. Install Prometheus Operator. You can follow up the steps in [Kube Prometheus](https://github.com/coreos/kube-prometheus) project.
+2. Then, enable the project to export the metrics by uncommenting the line `- ../prometheus` in the `config/default/kustomization.yaml` to allow 
+it creates the `ServiceMonitor` resource which is used to do this export. Following an example.
+
+```yaml
+# [PROMETHEUS] To enable prometheus monitor, uncomment all sections with 'PROMETHEUS'.
```

Comment:
As per ""testdata/project-v2/config/prometheus/monitor.yaml", it is configuring "ServiceMonitor" with "https" as port. Are you sure that this works without providing "tlsConfig" to pass certificates? This didn't work in my tests so I just wanted to verify. 

I would also remove "to allow" and just say: "It creates `ServiceMonitor` resource that allows Prometheus to scrape metrics from the controllers.".

Reply:
hi @draghuram, 

Please, be sure that you follow these instructions and use the https://github.com/coreos/kube-prometheus. The other repo suggested by you I understand that is no longer maintained. 

I'd recommend you check here: https://github.com/kubernetes-sigs/kubebuilder/issues/1244
As you see the metrics are exported.

Reply:
@camilamacedo86 Do we have a test covering this?

Reply:
We have a simple e2e test covering the install that we could improve.

Reply:
@camilamacedo86, I am indeed using v2. I will follow the instructions and see if I can check the metrics. Will keep you posted.

Reply:
Hi @camilamacedo86, I followed instructions in #1030 but I saw the following error in Prometheus for the metrics service endpoint:

"Get http://172.17.0.15:8443/metrics: net/http: HTTP/1.x transport connection broken: malformed HTTP response "\x15\x03\x01\x00\x02\x02""

Note that Prometheus was using "http" to access the end point. So I added the following 
```
    scheme: https
````
to "endpoints" of ServiceMonitor but then got this (expected) error:

"Get https://172.17.0.15:8443/metrics: x509: cannot validate certificate for 172.17.0.15 because it doesn't contain any IP SANs"

This is because the certificate used by the metrics service is not issued for the IP. I think we need to pass proper certificate info in "tlsConfig" (see https://coreos.com/operators/prometheus/docs/latest/api.html#tlsconfig). Either that or we need to enable no auth access to metrics end point.

Reply:
Hi @draghuram, 

Shows that your project has something different from the scaffold or it is something in the setup. I am unable to reproduce your scenario. For we troubleshooting it and identify what is missing I raised the issue: https://github.com/kubernetes-sigs/kubebuilder/issues/1253. There have all the steps that I just did with the master branch. Since it is out of the scope of this PR, let's move your concerns for there.  

c/c @mengqiy

Reply:
Sure, thanks.

Reply:
Closing it as solved.